fastplotlib/layouts/_frame/_ipywidget_toolbar.py

```diff
@@ -0,0 +1,297 @@
+            layout=Layout(width="auto"),
+            tooltip="panzoom controller",
+        )
+        self._maintain_aspect_button = ToggleButton(
+            value=True,
+            disabled=False,
+            description="1:1",
+            layout=Layout(width="auto"),
+            tooltip="maintain aspect",
+        )
+        self._maintain_aspect_button.style.font_weight = "bold"
+
+        self._y_direction_button = Button(
+            value=False,
+            disabled=False,
+            icon="arrow-up",
+            layout=Layout(width="auto"),
+            tooltip="y-axis direction",
+        )
+
+        self._record_button = ToggleButton(
+            value=False,
+            disabled=False,
+            icon="video",
+            layout=Layout(width="auto"),
+            tooltip="record",
+        )
+
+        self._add_polygon_button = Button(
+            value=False,
+            disabled=False,
+            icon="draw-polygon",
+            layout=Layout(width="auto"),
+            tooltip="add PolygonSelector"
+        )
+
+        widgets = [
+            self._auto_scale_button,
+            self._center_scene_button,
+            self._panzoom_controller_button,
+            self._maintain_aspect_button,
+            self._y_direction_button,
+            self._add_polygon_button,
+            self._record_button,
+        ]
+
+        if hasattr(self.plot, "_subplots"):
+            positions = list(product(range(self.plot.shape[0]), range(self.plot.shape[1])))
+            values = list()
+            for pos in positions:
+                if self.plot[pos].name is not None:
+                    values.append(self.plot[pos].name)
+                else:
+                    values.append(str(pos))
+
+            self._dropdown = Dropdown(
+                options=values,
+                disabled=False,
+                description="Subplots:",
+                layout=Layout(width="200px"),
+            )
+
+            self.plot.renderer.add_event_handler(self.update_current_subplot, "click")
+
+            widgets.append(self._dropdown)
+
+        self._panzoom_controller_button.observe(self.panzoom_handler, "value")
+        self._auto_scale_button.on_click(self.auto_scale_handler)
+        self._center_scene_button.on_click(self.center_scene_handler)
+        self._maintain_aspect_button.observe(self.maintain_aspect_handler, "value")
+        self._y_direction_button.on_click(self.y_direction_handler)
+        self._add_polygon_button.on_click(self.add_polygon)
+        self._record_button.observe(self.record_plot, "value")
+
+        # set initial values for some buttons
+        self._maintain_aspect_button.value = self.current_subplot.camera.maintain_aspect
+
+        if copysign(1, self.current_subplot.camera.local.scale_y) == -1:
+            self._y_direction_button.icon = "arrow-down"
+        else:
+            self._y_direction_button.icon = "arrow-up"
+
+        HBox.__init__(self, widgets)
```

Comment:
why this notation?

Reply:
Instantiate the subclass, I could do a super instead too

Reply:
changed to super